--- dashboard.jsp	2010-01-29 00:23:53.000000000 +0300
+++ dashboard.jsp	2010-01-29 00:55:14.000000000 +0300
@@ -92,7 +92,7 @@
             <tbody>
               <tr>
                 <td class="play" rowspan="2">
-                  <a href="#" onclick="callServer('<%= jmxURL.toExternalForm() + project.getProject() %>');">
+                  <a href="#" onclick="return callServer('<%= jmxURL.toExternalForm() + project.getProject() %>');">
                   </a>
                 </td>
                 <td class="left">
